scripts/symbol_to_entrez_id.py

```diff
@@ -0,0 +1,227 @@
+
+    unmatched_before = len(unmatched)
+    loc_map, unmatched = map_loc_genes(unmatched)
+    symbol_map = {**symbol_map, **loc_map}
+
+    if verbose:
+        print('-- Matched {} of {} genes'.format(
+            unmatched_before - len(unmatched), unmatched_before))
+
+    if len(unmatched) == 0:
+        return symbol_map
+
+    time.sleep(sleep_time)
+
+    if verbose:
+        print('Querying MyGene for aliases of {} unmatched genes:'.format(
+            len(unmatched)))
+
+    # then query for aliases of unmatched symbols
+    df_alias = mg.querymany(unmatched,
+                            scopes='alias',
+                            fields='symbol',
+                            species='human',
+                            verbose=False,
+                            as_dataframe=True)
+
+    # get rid of rows where the alias has already been matched
+    df_alias = df_alias.loc[~df_alias['symbol'].isin(matched)]
+
+    # duplicates are sorted in order of MyGene confidence score,
+    # so keep the most confident and drop others
+    #
+    # TODO: maybe revisit this and try to keep genes that match
+    # with TCGA data?
+    df_alias = df_alias.loc[~df_alias.index.duplicated(keep='first')]
+    df_alias = df_alias.loc[~df_alias['symbol'].duplicated(keep='first')]
+
+    if verbose:
+        num_genes, num_matched_genes = get_num_genes(df_alias)
+        print('-- Found aliases for {} of {} genes'.format(
+            num_matched_genes, len(unmatched)))
+
+    alias_map = query_to_map(df_alias, 'symbol', map_to_lists=True)
+    inverse_alias_map = invert_map(alias_map)
+
+    time.sleep(sleep_time)
+
+    if verbose:
+        print('Querying for alias entrez IDs:')
+
+    # and get entrez IDs of aliases
+    flat_aliases = [i for sl in alias_map.values() for i in sl]
+    df_inexact = mg.querymany(flat_aliases,
+                              scopes='symbol',
+                              fields='entrezgene',
+                              species='human',
+                              verbose=False,
+                              as_dataframe=True)
+    df_inexact = df_inexact[df_inexact['entrezgene'].notnull()]
+
+    if verbose:
+        num_genes, num_matched_genes = get_num_genes(df_inexact)
+        print('-- Matched {} of {} genes'.format(
+            num_matched_genes, num_genes))
+
+    inexact_map = query_to_map(df_inexact, 'entrezgene')
+    inexact_map = {inverse_alias_map[k]: v
+                    for k, v in inexact_map.items()}
+
+    symbol_map = fill_na({**symbol_map, **inexact_map}, symbols_list)
+
+    if verbose:
+        eids = list(symbol_map.values())
+        total_count = len(eids)
+        na_count = eids.count('N/A')
+        duplicates = get_list_duplicates(eids)
+        print('RESULTS: matched {} of {} genes ({} duplicate Entrez IDs)'.format(
+            total_count - na_count, total_count, len(duplicates)))
+
+    return symbol_map
+
+if __name__ == '__main__':
```

Comment:
Is this being treated like a unit test? If so, it may be preferable to have a separate file for testing functions.

Reply:
Yes - I agree, I think I'll create a separate PR later this week with some unit tests. Will remove this for now, it was mainly just useful during development.